src/AppInstallerCLICore/Workflows/MSStoreInstallerHandler.cpp

```diff
@@ -64,6 +73,70 @@ namespace AppInstaller::CLI::Workflow
 
             THROW_IF_FAILED(hr);
         }
+
+        HRESULT DownloadMSStorePackageFile(const MSStore::MSStoreDownloadFile& downloadFile, const std::filesystem::path& downloadDirectory, Execution::Context& context)

```

Comment:
It feels like there should be some progress and cancellation handling.

Reply:
DownloadPackageInstaller workflow task (within this method) handles progress, cancellation and retry for each download. I think we could improve in the future if this experience is not enough.